drivers/mtd/Kconfig

```diff
@@ -5,6 +5,88 @@
 # directory for more details.
 #
 
-config MODULE_MTD
+config HAVE_MTD_AT24CXXX
+    bool
+    depends on MODULE_AT24CXXX
+    select MODULE_MTD_AT24CXXX if MODULE_MTD
+    help
+        Indicates that a at24cxxx EEPROM MTD is present
+
+config HAVE_MTD_AT25XXX
+    bool
+    select MODULE_MTD_AT25XXX if MODULE_MTD
+    help
+        Indicates that a at25xxx SPI-EEPROM MTD is present
+
+config HAVE_MTD_NATIVE
+    bool
+    select MODULE_MTD_NATIVE if MODULE_MTD
+    help
+        Indicates that a native MTD is present.
+
+config HAVE_MTD_SD_SCARD
+    bool
+    select MODULE_MTD_SD_CARD if MODULE_MTD
+    help
+        Indicates that a sdcard MTD is present
+
+config HAVE_MTD_SPI_NOR
+    bool
+    select MODULE_MTD_SPI_NOR if MODULE_MTD
+    help
+        Indicates that a spi-nor MTD is present
+
+config HAVE_MTD_SPI_MCI
+    bool
+    select MODULE_MTD_MCI if MODULE_MTD
+    help
+        Indicates that a Multimedia Card Interface (MCI) MTD is present
+
+menuconfig MODULE_MTD
     bool "Memory Technology Device interface (MTD)"
     depends on TEST_KCONFIG
+
+if MODULE_MTD
+
+menu "MTD Interefaces"
+
+config MODULE_MTD_SPI_NOR
+    bool "MTD interface for SPI NOR Flash"
+    depends on HAS_PERIPH_SPI
+    select MODULE_PERIPH_SPI
+
+config MODULE_MTD_FLASHPAGE
+    bool "MTD interface for Flashpage"
+    depends on HAS_PERIPH_FLASHPAGE
+    depends on HAS_PERIPH_FLASHPAGE_PAGEWISE
+    select MODULE_PERIPH_FLASHPAGE
+    select MODULE_PERIPH_FLASHPAGE_PAGEWISE
+    help
+        Driver for internal flash devices implementing flashpage interface.
+
+config MODULE_MTD_NATIVE
+    bool "MTD native driver"
+    depends on NATIVE_OS_LINUX
+
+config MODULE_MTD_AT24CXXX
+    bool "MTD implementation for AT24CXXX"
+    depends on MODULE_AT24CXXX
+
+config MODULE_MTD_AT25XXX
+    bool "MTD implementation for AT25XXX"
+    depends on MODULE_AT25XXX
+
+config MODULE_MTD_MCI
+    bool "MTD interface for LPC23XX MCI"
+    depends on MODULE_MCI
```

Comment:
What about we add `depends on CPU_FAM_LPC23XX` here and `select MODULE_MCI`? Otherwise `HAVE_MTD_SPI_MCI` might select this with missing dependencies.

Reply:
makes sense thanks!